src/charm.py

```diff
@@ -71,6 +72,10 @@ def _on_install(self, _):
     def _on_config_changed(self, event):
         self._configure(event)
 
+    def _on_upgrade_charm(self, event):
+        self._patch_k8s_service()
+        self._configure(event)
```

Comment:
Note that this hook is simple thanks to pulling the config every time.
In alertmanager I store the hash and because of that my upgrade hook is more involved.
Just pointing out :)